main.py

```diff
@@ -60,6 +61,14 @@ def get(self):
         """Handles GET warmup requests."""
         pass
 
+class RedirectBasedOnLoginStatus(base.BaseHandler):
```

Comment:
Let's call this HomePageRedirectHandler, to keep with the convention of handlers having names that end with 'Handler'.

Also, small nit: can you add an extra newline above, to follow the convention in the code that two newlines separate all top-level classes?